src/mgmt/srv_target.c

```diff
@@ -506,22 +512,31 @@ tgt_vos_preallocate(uuid_t uuid, daos_size_t scm_size, int tgt_nr)
 
 		/** Align to 4K or locking the region based on the size will fail */
 		scm_size = D_ALIGNUP(scm_size, 1ULL << 12);
-		/**
-		 * Pre-allocate blocks for vos files in order to provide
-		 * consistent performance and avoid entering into the backend
-		 * filesystem allocator through page faults.
-		 * Use fallocate(2) instead of posix_fallocate(3) since the
-		 * latter is bogus with tmpfs.
-		 */
-		rc = fallocate(fd, 0, 0, scm_size);
-		if (rc) {
-			rc = daos_errno2der(errno);
-			D_ERROR(DF_UUID": failed to allocate vos file %s with "
-				"size: "DF_U64": "DF_RC"\n",
-				DP_UUID(uuid), path, scm_size, DP_RC(rc));
+		size = 1024ULL * 1024ULL * 1024ULL; /* Allocation chunk size: 1Gb */
+		off = 0;
+		do {
+			if (off + size > scm_size)
+				size = scm_size - off;
+			/**
+			 * Pre-allocate blocks for vos files in order to provide
+			 * consistent performance and avoid entering into the backend
+			 * filesystem allocator through page faults.
+			 * Use fallocate(2) instead of posix_fallocate(3) since the
+			 * latter is bogus with tmpfs.
+			 */
+			rc = fallocate(fd, 0, off, size);
+			if (rc) {
+				rc = daos_errno2der(errno);
+				D_ERROR(DF_UUID": failed to allocate vos file %s with "
+					"size: "DF_U64" of "DF_U64": "DF_RC"\n",
+					DP_UUID(uuid), path, off + size, scm_size, DP_RC(rc));
+				break;
+			}
+			usleep(1000);
```

Comment:
We probably need to sleep a bit less here since 1TB is already 1s. usleep(10) should be enough.

Reply:
I think short sleep like 10us will not give a chance to switch to other thread. It will just busy loop inside libc. I assume we need to sleep more than 100us to allow switch to other threads.

Reply:
well, may be you could use a lower tempo and force yielding with pthread_yield() ?

Reply:
Good suggestion!

Reply:
@NiuYawei do you think a pthread_yield()/sched_yield() here could have the same wrong/side effect to make xstream 0 giving up CPU?

Reply:
I think it's safe to call pthread_yield() or sched_yield() here, since this tgt_vos_preallocate() is only called by the dedicated fallocate pthread.

Reply:
According to my experiments this work not good enough. usleep works better for this needs.